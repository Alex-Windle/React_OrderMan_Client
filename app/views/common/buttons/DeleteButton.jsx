/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 */


// Module imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';


/**
 * @class DeleteButton
 * @description Button indicating a Delete action.
 */
export default class DeleteButton extends Component {
  constructor(props) {
    super(props);

    this._onClick = () => {
      const isSure = confirm('Are you sure? This is permanent.');
      if (isSure) {
        this.props.onClick();
      }
    };
  }

  render() {
    return (
      <button
        className="btn btn-danger"
        onClick={this._onClick}
      >
        Delete
      </button>
    );
  }
}


DeleteButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};


DeleteButton.defaultProps = {
  disabled: false,
};
