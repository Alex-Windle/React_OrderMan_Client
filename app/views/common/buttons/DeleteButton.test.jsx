/**
 * GENERAL NOTES
 * @author TalkRise <admin@talkrise.com>
 */


// Module imports
import React from 'react';
import renderer from 'react-test-renderer';


// Component imports
import DeleteButton from './DeleteButton';


describe('DeleteButton component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <DeleteButton onClick={jest.fn()} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
