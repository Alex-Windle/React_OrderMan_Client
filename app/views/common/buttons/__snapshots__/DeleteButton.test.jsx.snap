// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton component matches snapshot 1`] = `
<button
  className="btn btn-danger"
  onClick={[Function]}
>
  Delete
</button>
`;
